main/src/ca/uwaterloo/flix/language/phase/Deriver.scala

```diff
@@ -31,39 +33,43 @@ import ca.uwaterloo.flix.util.{InternalCompilerException, Validation}
   */
 object Deriver {
 
-  def run(root: KindedAst.Root)(implicit flix: Flix): Validation[KindedAst.Root, Nothing] = flix.phase("Deriver") {
-    val derivedInstances = root.enums.values.flatMap {
+  def run(root: KindedAst.Root)(implicit flix: Flix): Validation[KindedAst.Root, DerivationError] = flix.phase("Deriver") {
+    val derivedInstances = (traverse(root.enums.values) {
       enum => getDerivedInstances(enum, root)
+    }).map(_.flatten)
+
+    flatMapN(derivedInstances) {
```

Comment:
Here we shouldn't have to `flatMap`, just `map` and remove the `toSuccess` in the body.

Reply:
Ah! Thanks, didn't know about `map`.